src/utils/general.py

```diff
@@ -0,0 +1,25 @@
+import uuid
+
+
+def generate_project_input_paths(project_path, sources, targets):
+    paths = []
+
+    for source, target in zip(sources, targets):
+        paths.append(f"{project_path}/rules/rules-reviewed/{target}/{source}/tests/data")
+        # todo: make sure path exists before adding
+    return paths
+
+
+def generate_uuid():
+    return uuid.uuid4()
+
+
+def write_data_to_file(file_path, content):
+    with open(file_path, "w") as file:
+        file.write(content)
```

Comment:
The file needs to be closed.

Reply:
Using a **with open()** statement will automatically close the file once the block has completed

Reply:
Oversight. This looks good.